clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java

```diff
@@ -289,6 +293,11 @@ public void close() {
         };
     }
 
+    /**
+     * For uncompressed iterator, it is actually not worth skipping key / value / headers at all since
```

Comment:
nit: I think I would move this comment down to line 367. It feels out of context here.